feedhenry/fh-fhc
# FH-3746 - Change the cmds  eventalert * to follow the V3 standard and…

JIRA: https://issues.jboss.org/browse/FH-3746

**Following the changes**

- Move to V3 format and directory
- Add unit tests
- Improve output
- Change to use the same REST API of the NGUI

**Steps to Verify**

- Create a new project
- Go to the cloud app of this project
- Go to `Notifications` tab
- Click on Alerts ( this command is about this area )
- Create a new alert via the studio
- List all alerts via fhc 
Command: `fhc eventalert list --app=<app> --env=<environment>`
- Clone a alert
Command: `fhc eventalert clone --app=<app> --id=<id> --name=<name> --env=<environment>`
- Delete a alert
Command: `fhc eventalert delete --app=<app> --id=<id>  --env=<environment>`
- Read a alert
Command: `fhc eventalert read --app=<app> --id=<id> --env=<environment>`
- Update a alert **
Command: `fhc eventalert update --app=<app> --id=<id> --name=<name> --categories=<categories> --severities=<severities> --events=<events> --emails=<emails> --env=<environment>`
- Create a alert **
Command: `cmd : 'fhc eventalert create --app=<app> --name=<name> --categories=<categories> --severities=<severities> --events=<events> --emails=<emails> --env=<environment>`

**IMPORTANT** Use valid values for `--categories=APP_STATE,APP_ENVIRONMENT --severities=INFO,WARN,ERROR,FATAL --events=CRASHED`

If an invalid value is used the following error is found and the data is compromised as follows.

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert create 7gwbu7lepnxijuyhcxrqtodg --name=New22 --categories=APP_STATE,APP_ENVIRONMENT --severities=INFO,WARN,ERROR,FATAL --events=CRAD --emails=emails@test.com,emails2@test.com --env=dev
fhc ERR! String index out of range: -1 
fhc Command executed with error
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert list 7gwbu7lepnxijuyhcxrqtodg --env=dev 
{
  "error": "String index out of range: -1",
  "status": "error"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
``` 

A JIRA was open for the REST API check the informed values before save. The JIRA is: https://issues.jboss.org/browse/RHMAP-16656

**Following the local tests**

- To create:

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert create cnrpzc7ssngewe37parb5m7a --name=Test2 --categories=APP_STATE,APP_ENVIRONMENT --severities=INFO,WARN,ERROR,FATAL --events=CRASHED --emails=emails@test.com,emails2@test.com --env=dev --json
{
  "alertName": "Test2",
  "emails": "emails@test.com,emails2@test.com",
  "enabled": true,
  "env": "dev",
  "eventCategories": "APP_STATE,APP_ENVIRONMENT",
  "eventNames": "CRASHED",
  "eventSeverities": "INFO,WARN,ERROR,FATAL",
  "guid": "f3bcw4jwsioqid37gewqjeoc",
  "status": "ok",
  "sysCreated": "2017-07-06 18:23:13",
  "sysModified": "2017-07-06 18:23:13",
  "uid": "cnrpzc7ssngewe37parb5m7a"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

<img width="1150" alt="screen shot 2017-07-06 at 3 23 18 pm" src="https://user-images.githubusercontent.com/7708031/27926539-251c7ae4-625f-11e7-9da1-fc47fcadb8b7.png">

<img width="1261" alt="screen shot 2017-07-06 at 3 24 28 pm" src="https://user-images.githubusercontent.com/7708031/27926577-3ab518ac-625f-11e7-8ad5-fd12cb42e268.png">

- To List:

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert list cnrpzc7ssngewe37parb5m7a --env=dev --json
{
  "list": [
    {
      "alertName": "Test2",
      "emails": "emails@test.com,emails2@test.com",
      "enabled": true,
      "env": "dev",
      "eventCategories": "APP_STATE,APP_ENVIRONMENT",
      "eventNames": "CRASHED",
      "eventSeverities": "INFO,WARN,ERROR,FATAL",
      "guid": "f3bcw4jwsioqid37gewqjeoc",
      "sysCreated": "2017-07-06 18:23:13",
      "sysModified": "2017-07-06 18:23:13",
      "uid": "cnrpzc7ssngewe37parb5m7a"
    },
    {
      "alertName": "Test",
      "emails": "emails@test.com,emails2@test.com",
      "enabled": true,
      "env": "dev",
      "eventCategories": "APP_STATE,APP_ENVIRONMENT",
      "eventNames": "CRASHED",
      "eventSeverities": "INFO,WARN,ERROR,FATAL",
      "guid": "ndewinnsbpk25cdwsbj62cok",
      "sysCreated": "2017-07-06 18:22:54",
      "sysModified": "2017-07-06 18:22:54",
      "uid": "cnrpzc7ssngewe37parb5m7a"
    }
  ],
  "status": "ok"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```
<img width="1092" alt="screen shot 2017-07-06 at 3 25 32 pm" src="https://user-images.githubusercontent.com/7708031/27926628-6029c0b0-625f-11e7-9418-4dd4de5b9cc6.png">

- To delete:

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert delete cnrpzc7ssngewe37parb5m7a ndewinnsbpk25cdwsbj62cok --env=dev --json
{
  "alertName": "Test",
  "emails": "emails@test.com,emails2@test.com",
  "enabled": true,
  "env": "dev",
  "eventCategories": "APP_STATE,APP_ENVIRONMENT",
  "eventNames": "CRASHED",
  "eventSeverities": "INFO,WARN,ERROR,FATAL",
  "guid": "ndewinnsbpk25cdwsbj62cok",
  "status": "ok",
  "sysCreated": "2017-07-06 18:22:54",
  "sysModified": "2017-07-06 18:22:54",
  "uid": "cnrpzc7ssngewe37parb5m7a"
}
Camilas-MacBook-Pro:fh-fhc cmacedo$ 
```

```
Camilas-MacBook-Pro:fh-fhc cmacedo$ ./bin/fhc.js eventalert delete cnrpzc7ssngewe37parb5m7a f3bcw4jwsioqid37gewqjeoc --env=dev
Event alerts with the ID 'f3bcw4jwsioqid37gewqjeoc' for the app 'cnrpzc7ssngewe37parb5m7a' into the env 'dev' deleted successfully.
```

- To update: 

<img width="1135" alt="screen shot 2017-07-06 at 3 44 10 pm" src="https://user-images.githubusercontent.com/7708031/27927411-0a914b0c-6262-11e7-9fb4-41bf5d81a7a1.png">

- To clone:

<img width="1077" alt="screen shot 2017-07-06 at 3 45 59 pm" src="https://user-images.githubusercontent.com/7708031/27927475-400b327a-6262-11e7-8a38-ed8f33fad83a.png">
